src/api/search.js

```diff
@@ -1,171 +1,125 @@
-      chains: chains.join(','),
-      categories: categories.join(','),
-      groups: groups.join(','),
-      limit: Math.min(limit, 100),
     };
 
-    Http.request('GET', 'v1/search/places', queryParams,
-      (status, response) => {
-        callback(status, response ? response.places : null);
-        return;
-      }
-    );
+    const response = await Http.request('GET', 'v1/search/places', params);
+
+    return response.places;
   }
 
-  static searchGeofences(
-    {
+  static async searchGeofences(searchOptions={}) {
+    if (!searchOptions.near) {
+      const { latitude, longitude } = await Navigator.getCurrentPosition();
+      searchOptions.near = { latitude, longitude };
+    }
+
+    let {
+      near,
       radius,
       tags,
       limit,
-    },
-    callback,
-  ) {
-    Navigator.getCurrentPosition((status, { latitude, longitude }) => {
-      if (status !== STATUS.SUCCESS) {
-        callback(status);
-        return;
-      }
-
-      this.searchGeofencesNear(
-        {
-          near: { latitude, longitude },
-          radius,
-          tags,
-          limit,
-        },
-        (status, geofences) => {
-          callback(status, geofences);
-          return;
-        }
-      );
-    });
-  }
+    } = searchOptions;
 
-  static searchGeofencesNear(
-    {
+    if (near) {
+      near = `${near.latitude},${near.longitude}`;
+    }
+    if (tags) {
+      tags = tags.join(',');
+    }
+
+    limit = getLimit(limit);
+
+    const params = {
       near,
       radius,
       tags,
       limit,
-    },
-    callback,
-  ) {
-    const queryParams = {
-      near: `${near.latitude},${near.longitude}`,
-      radius,
-      tags: tags.join(','),
-      limit: Math.min(limit, 100),
     };
 
-    Http.request('GET', 'v1/search/geofences', queryParams,
-      (status, response) => {
-        callback(status, response ? response.geofences : null);
-        return;
-      }
-    );
-  }
+    const response = await Http.request('GET', 'v1/search/geofences', params);
 
-  static autocomplete(
-    {
-      query,
-      limit,
-    },
-    callback
-  ) {
-    Navigator.getCurrentPosition((status, { latitude, longitude }) => {
-      if (status !== STATUS.SUCCESS) {
-        callback(status);
-        return;
-      }
-
-      this.autocompleteNear(
-        {
-          query,
-          near: { latitude, longitude },
-          limit,
-        },
-        (status, addresses) => {
-          callback(status, addresses);
-          return;
-        }
-      )
-    });
+    return response.geofences;
   }
 
-  static autocompleteNear(
-    {
+
+  static async autocomplete(searchOptions={}) {
+    // NOTE: I'm not sure how caching works with getCurrentPosition,
+    // but this could be problematic here if needs to compute each time.
+    // We could add a cache: true flag to retreive previous value
+    if (!searchOptions.near) {
+      const { latitude, longitude } = await Navigator.getCurrentPosition();
```

Comment:
looks like you can give the navigator a `maximumAge` for freshness of the position - https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition